pkg/kubelet/pod_workers.go

```diff
@@ -180,7 +180,13 @@ func (p *podWorkers) managePodLoop(podUpdates <-chan UpdatePodOptions) {
 		}
 		if err != nil {
 			glog.Errorf("Error syncing pod %s (%q), skipping: %v", update.Pod.UID, format.Pod(update.Pod), err)
-			p.recorder.Eventf(update.Pod, v1.EventTypeWarning, events.FailedSync, "Error syncing pod, skipping: %v", err)
+			// if we failed sync, we throw more specific events for why it happened.
```

Comment:
We don't, or at least not in `syncPod()` in kubelet.go. There are errors if the pod is not runnable or network is not ready, and I only checked a few.  This lets you aggregates event, yes, but may actually lose the information that user wants to know.